auto_submit/lib/requests/check_pull_request.dart

```diff
@@ -76,11 +79,47 @@ class CheckPullRequest extends RequestHandler {
     return filesIsEmpty;
   }
 
-  Future<bool> _removeLabel() async {
-    // TODO(Kristin): Implement the logic to remove the label. https://github.com/flutter/flutter/issues/99877
+  /// Removes the 'autosubmit' label if this PR should not be merged.
+  ///
+  /// Returns true if we successfully remove the label.
+  Future<bool> _removeLabel(
+      _AutoMergeQueryResult queryResult, GithubService gitHub, RepositorySlug slug, String label) async {
+    log.info('This pull request is not suitable for automatic merging in its '
+        'current state.');
+
+    if (!queryResult.hasApprovedReview && queryResult.changeRequestAuthors.isEmpty) {
+      log.info('- Please get at least one approved review if you are already '
+          'a member or two member reviews if you are not a member before re-applying this '
+          'label. __Reviewers__: If you left a comment approving, please use '
+          'the "approve" review action instead.');
+    }
+    for (String? author in queryResult.changeRequestAuthors) {
+      log.info('- This pull request has changes requested by @$author. Please '
+          'resolve those before re-applying the label.');
+    }
+    for (_FailureDetail detail in queryResult.failures) {
+      log.info('- The status or check suite ${detail.markdownLink} has failed. Please fix the '
+          'issues identified (or deflake) before re-applying this label.');
+    }
+    if (queryResult.emptyChecks) {
+      log.info('- This commit has no checks. Please check that ci.yaml validation has started'
+          ' and there are multiple checks. If not, try uploading an empty commit.');
+    }
+    if (queryResult.isConflicting) {
+      log.info('- This commit is not mergeable and has conflicts. Please'
+          ' rebase your PR and fix all the conflicts.');
+    }
+    final bool result = await gitHub.removeLabel(slug, queryResult.number, 'autosubmit');
```

Comment:
Save `autosubmit` in a variable like in the config.dart to make it accessible from multiple locations. This will also make it easy to manage.

Reply:
Good suggestion, thanks Keyong, I will add the label to config.